genesyscloud/resource_genesyscloud_routing_settings.go

```diff
@@ -0,0 +1,297 @@
+}
+
+func readRoutingSettingsContactCenter(d *schema.ResourceData, routingAPI *platformclientv2.RoutingApi) diag.Diagnostics {
+	contactcenter, resp, getErr := routingAPI.GetRoutingSettingsContactcenter()
+	if getErr != nil {
+		if isStatus404(resp) {
+			return nil
+		}
+		return diag.Errorf("Failed to read Contact center for routing setting %s: %s\n", d.Id(), getErr)
+	}
+
+	if contactcenter == nil {
+		d.Set("contactcenter", nil)
+		return nil
+	}
+
+	contactSettings := make(map[string]interface{})
+
+	if contactcenter.RemoveSkillsFromBlindTransfer != nil {
+		contactSettings["remove_skills_from_blind_transfer"] = *contactcenter.RemoveSkillsFromBlindTransfer
+	}
+
+	d.Set("contactcenter", []interface{}{contactSettings})
+	return nil
+}
+
+func readRoutingSettingsTranscription(d *schema.ResourceData, routingAPI *platformclientv2.RoutingApi) diag.Diagnostics {
+	transcription, resp, getErr := routingAPI.GetRoutingSettingsTranscription()
+	if getErr != nil {
+		if isStatus404(resp) {
+			return nil
+		}
+		return diag.Errorf("Failed to read Contact center for routing setting %s: %s\n", d.Id(), getErr)
+	}
+
+	if transcription == nil {
+		d.Set("transcription", nil)
+		return nil
+	}
+
+	transcriptionSettings := make(map[string]interface{})
+
+	if transcription.Transcription != nil {
+		transcriptionSettings["transcription"] = *transcription.Transcription
+	}
+	if transcription.TranscriptionConfidenceThreshold != nil {
+		transcriptionSettings["transcription_confidence_threshold"] = *transcription.TranscriptionConfidenceThreshold
+	}
+	if transcription.LowLatencyTranscriptionEnabled != nil {
+		transcriptionSettings["low_latency_transcription_enabled"] = *transcription.LowLatencyTranscriptionEnabled
+	}
+	if transcription.ContentSearchEnabled != nil {
+		transcriptionSettings["content_search_enabled"] = *transcription.ContentSearchEnabled
+	}
+
+	d.Set("transcription", []interface{}{transcriptionSettings})
+	return nil
+}
+
+func updateContactCenter(d *schema.ResourceData, routingAPI *platformclientv2.RoutingApi) diag.Diagnostics {
+	var removeSkillsFromBlindTransfer bool
+
+	if contactCenterConfig := d.Get("contactcenter"); contactCenterConfig != nil {
+		if contactCenterList := contactCenterConfig.([]interface{}); len(contactCenterList) > 0 {
+			contactCenterMap := contactCenterList[0].(map[string]interface{})
+
+			if contactCenterMap["remove_skills_from_blind_transfer"] != nil {
+				removeSkillsFromBlindTransfer = contactCenterMap["remove_skills_from_blind_transfer"].(bool)
+			}
+			_, err := routingAPI.PatchRoutingSettingsContactcenter(platformclientv2.Contactcentersettings{
+				RemoveSkillsFromBlindTransfer: &removeSkillsFromBlindTransfer,
+			})
+
+			if err != nil {
+				return diag.Errorf("Failed to update Contact center for routing setting %s: %s\n", d.Id(), err)
+			}
+		}
+	}
+	return nil
+}
+
+func updateTranscription(d *schema.ResourceData, routingAPI *platformclientv2.RoutingApi) diag.Diagnostics {
+	var transcription string
```

Comment:
Just spotted this. Usually, we want to give variables as little scope as possible so the best thing would be to declare and initialize these only where they are used. So on line 272 it would be `transcription := transcriptionMap["transcription"].(string)` for example.